### PR TITLE
Fix async actions propagation in analytics middleware

### PR DESCRIPTION
**Pull request by PPiedel**

### CODE REVIEW

File: ui/client/analytics/AnalyticsMiddleware.js [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
+  const analyticsSettings = _.get(state, 'settings.analyticsSettings')
+  if (!_.isEmpty(analyticsSettings) && actionTracked(action)) {
     analytics(analyticsSettings).sendEvent(action.tracking)
   }
-
   return next(action)
-};
+}
 
 function actionTracked(action) {
```
Comment (lciolecki) [inline ui/client/analytics/AnalyticsMiddleware.js]: Simply `actionTracked = (action) =>  !_.isEmpty(action) && !_.isEmpty(action.tracking)`

File: ui/client/analytics/AnalyticsMiddleware.js [hunk 2]
```diff
@@ -1,18 +1,21 @@
 import Analytics from "./Analytics"
 
-export const analyticsMiddleware = (store) => next => action => {
-  const state = !_.isEmpty(store) ? store.getState() : null
-  const analyticsSettings = _.get(state, 'settings.analyticsSettings')
+export const analyticsMiddleware = ({dispatch, getState}) => next => action => {
+  if (typeof action === 'function') {
```
Comment (lciolecki) [inline ui/client/analytics/AnalyticsMiddleware.js]: Simply: `https://lodash.com/docs/4.17.15#isFunction`

File: ui/client/analytics/AnalyticsMiddleware.js (hunk 1 above)
Comment (PPiedel) [reply ui/client/analytics/AnalyticsMiddleware.js]: Done

File: ui/client/analytics/AnalyticsMiddleware.js (hunk 2 above)
Comment (PPiedel) [reply ui/client/analytics/AnalyticsMiddleware.js]: Done

**Pull request merged**